packages/http/src/types.ts

```diff
@@ -325,11 +325,29 @@ export interface RoutePath {
   shared: boolean;
 }
 
-export type StatusCode = `${number}` | "*";
 export interface HttpOperationResponse {
+  /** @deprecated use {@link statusCodes} */
+  // eslint-disable-next-line deprecation/deprecation
   statusCode: StatusCode;
+
+  /**
+   * Status code or range of status code for the response.
+   */
+  statusCodes: HttpStatusCodeRange | number | "*";
```

Comment:
Do we usually add and deprecate in same version?

It causes nightly build fail, as our build checks no deprecate API called.
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3150653&view=logs&j=ca395085-040a-526b-2ce8-bdc85f692774&t=cf4477f9-109f-5399-1612-bcbe5909c9f2

However there is no way to fix in emitter, as `statusCodes` is not in stable, and dev already deprecated `statusCode`.

PS: hope no service uses it right now. Java language code does not support status code range. So if service write 200-299, we likely had to generate `{200, 201, 202, 203, ..., 299}` in code.

Reply:
PS: hope no service uses it right now. Java language core does not support status code range. So if service write `@minValue(200) @maxValue(299)`, we likely had to generate `@ExpectedResponses({200, 201, 202, 203, ..., 299})` in code, in near future.
[ExpectedResponses](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/annotation/ExpectedResponses.java)
[UnexpectedResponseExceptionType](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/annotation/UnexpectedResponseExceptionType.java)

Reply:
we did this so it is not a breaking change, you can suppress teh warning with `//eslint-disable deperecation/deprecation` I think , that's how we do it in typespec.

Reply:
and yeah no service is using it at of now as we have not released. I also probably don't expect much azure services to use that if any.

Reply:
> we did this so it is not a breaking change, you can suppress teh warning with `//eslint-disable deperecation/deprecation` I think , that's how we do it in typespec.

Got it. Thanks. Going need to remind to remove it after next typespec release (though that should be near).